doc/null_safety_and_required_props.md

```diff
@@ -0,0 +1,411 @@
+# Null safety and required props
```

Comment:
GitHub's "rich diff" viewer apparently doesn't render [alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) very well, so I'd recommend viewing this file directly: https://github.com/Workiva/over_react/blob/null-safety-docs/doc/null_safety_and_required_props.md